lib/bundler/metrics.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require "time"
+require "securerandom"
+require "digest"
+
+module Bundler
+  class Metrics
+    attr_accessor :metrics_hash, :path
+
+    def initialize
+      @path = Pathname.new(File.join(Bundler.user_home.join(".bundle"), "metrics.yml"))
+      add_metrics
+    end
+
```

Comment:
Let's create a method here that we can specifically use to record a single metric. Something like:

```
def record(entry = {})
  entry["timestamp"] = Time.now.utc.iso8601
  @metrics.push(entry)
end
```

Reply:
I've taken a slightly different approach to avoid duplicating code (creating an entry locally in every `some_command.rb`), I'd like to hear your opinion about it.